ADAL/src/ui/ADWebAuthController.m

```diff
@@ -125,7 +125,7 @@ + (void)startWithRequest:(ADRequestParameters *)requestParams
     webviewConfig.loginHint = requestParams.identifier.userId;
     webviewConfig.promptBehavior = [ADAuthenticationContext getPromptParameter:promptBehavior];
     
-    webviewConfig.extraQueryParameters = [self.class dictionaryFromQueryString:requestParams.extraQueryParameters];
+    webviewConfig.extraQueryParameters = [self.class dictionaryFromQueryString:requestParams.extraQueryParameters.msidWWWFormURLDecode];
```

Comment:
Will our helper `msidDictionaryFromWWWFormURLEncodedString` work for this case? :)

Reply:
Current implementation is more robust and handles cases where entire string is encoded. Agreed offline to keep as is.